curriculum/challenges/english/25-front-end-development/lecture-importance-of-accessibility-and-good-html-structure/672a54dff9dc439089f1a219.md

```diff
@@ -10,6 +10,47 @@ dashedName: what-is-the-aria-live-attribute
+
+Common uses of live regions include messages that are displayed after an action has been taken (such as an error message or confirmation), content that updates periodically (such as a ticker, marquee, or countdown timer), or general status messages (such as updates about a process).
+
+Because the reading focus for screen readers can only be at one place at a time, screen reader users will not notice a content change if their focus is on another part of the page. Live regions allow screen reader users to be automatically notified of changes that happen on the page in real time. Without a live region, screen reader users might miss important content updates available to sighted users, since a sighted user has the ability to scan the entire page. 
+
+There are three possible values for this attribute, based on the priority of the information.
+If you set `aria-live` to the value `assertive` that means that the update is very important. It has the highest priority, so the user should be notified immediately.
+This means that the screen reader will interrupt any announcement it is currently making to announce the content change in the live region. Such interruptions can be extremely disruptive, so the `assertive` value should only be used for time-sensitive or critical notifications.
+
+```html
+<div aria-live="assertive">
+  <p>Your session will expire in 30 seconds.</p>
+</div>
+```
+
+The next value in order of priority is `polite`.
+This value means that the update is not  urgent, so the screen reader can wait until any  current announcement is finished or until the user stops typing before announcing the update. Most live regions will use the `polite` value.
+
+```html
+<div aria-live="polite">
+  <p>File successfully uploaded</p>
+</div>
+```
+
+The lowest priority value for `aria-live` is `off` which means that the update will not be announced  unless the content is in an element that currently has keyboard focus. Realistically, this value is almost never used as the use case is very narrow and it is not implemented consistently (if at all) across screen readers. If you need live regions, plan on using `polite` for everything except critical messages that need `assertive`.
+
+It's also important to note that the `aria-live` attribute is not required if the updated information is contained in an element with an ARIA role of `alert`, `log`, `marquee`, `status`, or `timer`, as these roles already have default `aria-live` values. But the default value can be changed by explicitly setting `aria-live` on the element.
+
+Choosing the right `aria-live` value depends on the priority of the updated information.
+If the updates are urgent, you should notify the user immediately with `assertive`. But you should use this only if the updates are really urgent because sudden interruptions can potentially disorient users and affect the user experience.
+If the update can wait until the current task is finished, you should use `polite` instead.
```

Comment:
```suggestion
Choosing the right `aria-live` value depends on the priority of the updated information.

If the updates are urgent, you should notify the user immediately with `assertive`. But you should use this only if the updates are really urgent because sudden interruptions can potentially disorient users and affect the user experience.

If the update can wait until the current task is finished, you should use `polite` instead.
```